### DISCUSSION TITLE
Selective refinement of mesh size near the grain boundaries

### DISCUSSION THREAD

**Discussion by hrushi0301** [neperfepx/neper#134] [Q&A] [answered]
Dear all,

I have been working with Neper for quite some time now and it is a great platform!  I have been using the following set of commands to refine the mesh near the crack surface of my notched bar:

```
$ neper -T -n 50 -morpho gg -domain "cube(1.2,0.1,1.2)" -transform "cut(ecylinder(0.6,0,0,0,1,0,0,0,1,1,0,0,0.15,0.01))" -regularization 1 -fmax 10 -mloop 2 -o notched_plate
$ neper -M notched_plate.tess -rclface "(domface==7)?0.3:3" -pl 1.5 -meshqualmin 1 -format inp -o notched_plate_mesh
```

I would like to know if it is possible to selectively refine the mesh size near the individual grain boundaries as well?

Thanks.

![notched_plate_3](https://user-images.githubusercontent.com/68965204/110411814-2ef05080-80b1-11eb-9e82-d37c6eb1c035.png)

![notched_plate_mesh_3](https://user-images.githubusercontent.com/68965204/110411821-3152aa80-80b1-11eb-9d13-f000968aff8d.png)

**Comment by rquey** [comment, marked as answer]
If you mean getting a finer mesh at grain boundaries than in the grain interiors, for all grains, then no, it is currently not possible.

**Reply by hrushi0301** [reply]
I see. Thanks a lot for the reply!
